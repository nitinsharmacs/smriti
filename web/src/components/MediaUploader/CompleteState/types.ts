export interface CompleteStateProps {
  targetUploads: number;
  achievedUploads: number;
  previews: string[]; // thumbnails
  onComplete: () => void;
}
