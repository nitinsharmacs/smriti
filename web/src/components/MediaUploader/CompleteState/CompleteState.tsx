import Modal from 'src/components/MediaUploader/Modal/Modal';
import Button from 'src/components/Button/Button';

import './completestate.css';
import type { CompleteStateProps } from 'src/components/MediaUploader/CompleteState/types';

const CompleteState = ({
  achievedUploads,
  targetUploads,
  previews,
  onComplete,
}: CompleteStateProps) => {
  return (
    <Modal>
      <div className='upload-complete-modal'>
        <div className='upload-complete-content'>
          <h4>
            Upload completed
            {achievedUploads < targetUploads ? ' partially' : ''}
          </h4>
          <div className='upload-summary'>
            {achievedUploads} of {targetUploads} files uploaded successfully.
            <br />
            Please click <b>Complete</b> to store permanently.
          </div>
          <div className='complete-controls'>
            <Button variant='contained' onClick={onComplete}>
              Complete
            </Button>
          </div>
        </div>
        {/* TODO:
         * - preview can be improved with multiple images.
         * - preview can be clickable that opens a screen with all the
         * uploaded images.
         */}
        <div className='upload-preview'>
          <figure>
            <img src={previews[0]} alt='upload-preview' />
          </figure>
        </div>
      </div>
    </Modal>
  );
};

export default CompleteState;
