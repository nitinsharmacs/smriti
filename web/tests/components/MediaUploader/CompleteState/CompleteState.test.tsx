import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import CompleteState from 'src/components/MediaUploader/CompleteState/CompleteState';
import { describe, expect, it, vi } from 'vitest';

describe('CompleteState', () => {
  it('should render complete state', () => {
    const { container } = render(
      <CompleteState
        targetUploads={2}
        achievedUploads={2}
        previews={['/preview']}
        onComplete={function (): void {
          throw new Error('Function not implemented.');
        }}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render partially complete state', () => {
    const { container } = render(
      <CompleteState
        targetUploads={2}
        achievedUploads={1}
        previews={['/preview']}
        onComplete={function (): void {
          throw new Error('Function not implemented.');
        }}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should invoke complete', async () => {
    const completeMock = vi.fn();
    render(
      <CompleteState
        targetUploads={2}
        achievedUploads={1}
        previews={['/preview']}
        onComplete={completeMock}
      />
    );

    const completeBtn = screen.getByRole('button');

    await userEvent.click(completeBtn);

    expect(completeMock).toHaveBeenCalledOnce();
  });
});
