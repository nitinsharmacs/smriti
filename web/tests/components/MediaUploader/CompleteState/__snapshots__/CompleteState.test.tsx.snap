// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompleteState > should render complete state 1`] = `
<div>
  <div
    class="upload-txn-modal"
  >
    <div
      class="upload-complete-modal"
    >
      <div
        class="upload-complete-content"
      >
        <h4>
          Upload completed
        </h4>
        <div
          class="upload-summary"
        >
          2
           of 
          2
           files uploaded successfully.
          <br />
          Please click 
          <b>
            Complete
          </b>
           to store permanently.
        </div>
        <div
          class="complete-controls"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary contained-btn css-74d805-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Complete
          </button>
        </div>
      </div>
      <div
        class="upload-preview"
      >
        <figure>
          <img
            alt="upload-preview"
            src="/preview"
          />
        </figure>
      </div>
    </div>
  </div>
</div>
`;

exports[`CompleteState > should render partially complete state 1`] = `
<div>
  <div
    class="upload-txn-modal"
  >
    <div
      class="upload-complete-modal"
    >
      <div
        class="upload-complete-content"
      >
        <h4>
          Upload completed
           partially
        </h4>
        <div
          class="upload-summary"
        >
          1
           of 
          2
           files uploaded successfully.
          <br />
          Please click 
          <b>
            Complete
          </b>
           to store permanently.
        </div>
        <div
          class="complete-controls"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary contained-btn css-74d805-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Complete
          </button>
        </div>
      </div>
      <div
        class="upload-preview"
      >
        <figure>
          <img
            alt="upload-preview"
            src="/preview"
          />
        </figure>
      </div>
    </div>
  </div>
</div>
`;
